internal/hardware/admission_mac.go

```diff
@@ -0,0 +1,70 @@
+package hardware
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/tinkerbell/tink/api/v1alpha2"
+	"github.com/tinkerbell/tink/internal/hardware/internal"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+func (a *Admission) validateMACs(hw *v1alpha2.Hardware) admission.Response {
```

Comment:
Could a cel validation be used instead? `+kubebuilder:validation:XValidation:rule=` https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/

Reply:
I don't think we can run regex using the `XValidation` rule - I can't find anything in the link or other resources that shows how.

Even if it could, I'm not sure it would work for _map keys_. We have CEL validation configured on the map key type which I'd hoped would translate to applying the regex, but it seems the open API spec supported by the `CustomResourceDefinition` kind doesn't support validation of map keys (not that I've found yet anyway).

Reply:
> Thanks @chrisdoherty4. Question, how is this deployed?

Great question.

This will be registered with the new controller manager component (we already have a manager under `/internal/controllers`). The manager is a controller-runtime thing that handles initialization of controllers, webhooks, clients and has responsibilities such as leader election etc. 

The `Admission` object satisfies the [`Handler`](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.6/pkg/webhook/admission#Handler) interface that is consumed by the [`Webhook`](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.6/pkg/webhook/admission#Webhook) data structure. The `Webhook` is [registerable](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.6/pkg/webhook/admission#Webhook.ServeHTTP), as an `http.Handler`, with the [`webhook.Server`](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.6/pkg/webhook#Server.Register).